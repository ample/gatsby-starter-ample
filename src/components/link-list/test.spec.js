import React from "react"
import renderer from "react-test-renderer"

import LinkList from "."

import horizontalFixture from "./__fixtures__/horizontal.json"
import verticalFixture from "./__fixtures__/vertical.json"

describe("LinkList", () => {
  it("renders horizontal lists correctly", () => {
    const tree = renderer.create(<LinkList links={horizontalFixture} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders vertical lists correctly", () => {
    const tree = renderer
      .create(<LinkList heading={verticalFixture.label} links={verticalFixture.links} vertical />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
