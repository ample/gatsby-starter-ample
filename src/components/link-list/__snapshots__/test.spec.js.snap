// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList renders horizontal lists correctly 1`] = `
<ul
  className="cl_link_list"
>
  <li>
    <a
      href="/"
    >
      Vestibulum
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Pellentesque venenatis
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Fusce
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Etiam lacinia dolor
    </a>
  </li>
  <li>
    <span
      className="dropdown"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="dropdown_trigger"
      >
        Suspendisse semper
      </span>
      <ul
        className="dropdown_menu"
      >
        <li>
          <a
            href="/"
          >
            Vestibulum vitae
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="noopener"
            target="_blank"
          >
            Pellentesque tempor venenatis
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="noopener"
            target="_blank"
          >
            Fusce molestie
          </a>
        </li>
        <li>
          <a
            href="#"
            rel="noopener"
            target="_blank"
          >
            Etiam in lacinia dolor
          </a>
        </li>
      </ul>
    </span>
  </li>
  <li>
    <a
      className="button button"
      rel="noopener"
      target="_blank"
    />
  </li>
</ul>
`;

exports[`LinkList renders vertical lists correctly 1`] = `
<ul
  className="cl_link_list cl_link_list_is_vertical"
>
  <li>
    <strong>
      Quisque
    </strong>
  </li>
  <li>
    <a
      href="/"
    >
      Vestibulum vitae
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Pellentesque tempor venenatis
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Fusce molestie
    </a>
  </li>
  <li>
    <a
      href="#"
      rel="noopener"
      target="_blank"
    >
      Etiam in lacinia dolor
    </a>
  </li>
</ul>
`;
